blueprints/layered-profile.md

```diff
@@ -0,0 +1,347 @@
+
+With Layered Profiles, hundreds of profiles become a few dozen, each representing a logical group. For example, a server might have the profiles, in order:
+1. EDGE
+2. AMIGA_123
+3. CDN_FOO
+4. DATACENTER_ABC
+5. ATS_714
+6. Custom_Hack_42
+
+### Traffic Portal Impact
+
+1. A UI to view all parameters currently applied to a server, as well as the profile each parameter came from. A new page, linked from Server pages. For eg:
+
+| Name                                             |Config File               | Value                                | Profile   |
+|--------------------------------------------------|--------------------------|--------------------------------------|-----------|
+| location                                         | url_sig_myds.config      | /opt/trafficserver/etc/trafficserver | EDGE      |
+| Drive_Prefix                                     | storage.config           | /dev/sd                              | AMIGA_123 |
+| error_url                                        | url_sig_myotherds.config | 403                                  | EDGE      |
+| CONFIG proxy.config.exec_thread.autoconfig.scale | records.config           | FLOAT 1.5                            | ATS_714   |
+
+2. A UI change to add, remove, and reorder (sortable list) profiles for Servers, on the existing Server pages.
+3. Filtering based on Profiles will also need to be updated to take into account the plurality of Profiles.
+
+### Traffic Ops Impact
+
+- Traffic Ops will need to add the logic to the below-mentioned existing API endpoints, in order to show/create/update/delete a sorted list of profiles for server.
+
+- `/servers/  GET, POST`
+- `/servers/{id} PUT, DELETE`
+- `/servers/details`
+
+#### REST API Impact
+
+- No new endpoints are required
+
+**Existing Endpoints**
+
+- Modify JSON request and response for existing servers endpoints.
+
+#### API 5.0 GET
+- JSON **response** with the proposed change will look as follows:
+`/servers?id=5`
+```JSON
+{ 
+  "id": 5,
+  ⋮
+  "requested": {
+    "layeredProfiles": [
+      "MID"
+    ],
+    ⋮
+  }
+}
+```
+
+`/servers`
+```JSON
+{ 
+  "requested": {
+    "layeredProfiles": [
+      "MID"
+    ],
+    ⋮
+  },
+  ⋮
+}
+```
+
+#### API 5.0 POST/PUT
+JSON **request** with the proposed change will look as follows:
+
+`POST /servers `
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+`PUT /servers/5`
+```JSON
+{
+    "cachegroupId": 6,
+    "cdnId": 2,
+    ⋮
+    "interfaces": [
+      ⋮
+    ],
+    ⋮
+    "profileId": [10, 11, 15],
+    ⋮
+}
+```
+
+returns a **400** **response** with
+
+```JSON
+{
+  "alerts":[
+   { 
+     "text":"cannot associate multiple profiles to a server with current API version but can be done via 5.0+",
+     "level":"error"
+   }
+]}
+```
+
+#### API 6.0 GET
```

Comment:
I think this section should be 5.0? And all of my comments above apply in turn to each endpoint here.

Reply:
I believe the plan was to make the `profileId` -> `profiles: []` data type change in API 4.0 since it's currently unstable. That is one of the 3ish breaking changes we currently had planned for the unstable API 4.0, meaning we wouldn't need an API 5.0 for this.

Reply:
Brennan explained above comment over chat.